scripts/combine/setupCombineWMass.py

```diff
@@ -243,7 +252,8 @@ def main(args):
                 passToFakes=passSystToFakes,
                 systNameReplace=nameReplace,
                 scale=scale,
-                splitGroup=splitGroupDict
+                splitGroup=splitGroupDict,
+                silentCheckOtherCharge=silentCheckOtherCharge
```

Comment:
I would propose to invert the logic here.

ie have a parameter "doChecksByCharge" which is False by default.  Ie one has to explicitly turn the extra checks on.  (Ie assume by default that a variation may not affect the "other" charge, and only impose this where it's explicitly known to be that way.)

Reply:
ok, but doesn't it contradict the purpose of the check? Usually you would like the check active for anything, and here we disable it because we know it is a false positive in some cases.
Otherwise if sanity checks become optional nobody would ever run them, and thus you'll never see problems until you have already discovered them independently (but at that point you just fix them, you don't add the check back)

Reply:
To put it differently, if you set the default check to be charge inclusive using the full TH3, then if the efficiency cases you'll end up having 95 % of bins or more equal to nominal in any case (50% from the whole wrong charge and the remaining almost 50% because the efficiency variations only modify 1 eta strip of the whole template), so at that point you have to fine tune the threshold to issue a warning, for example if it is even > 95% I expect it to always trigger regardless the "by charge" or "charge inclusive" logic

Reply:
One clarification, the current checks are already silent by default when the charge id does not appear in the nuisance name, and it appears there only when the syst tensor has the reco charge in its axis. So here the silentCheckOtherCharge is meant to disable it also when the charge is present, but one wants to skip the check because false positives are expected to be there for the wrong charge